### PR TITLE
Support manual optimization step profiling without a trainer reference

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Gets rid of a messy `strategy.lightning_module.trainer.profiler` reference in the LightningOptimizer. This will enable #11452

Part of #7315

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [n/a] Did you make sure to **update the documentation** with your changes? (if necessary)
- [n/a] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [n/a] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

cc @borda @justusschock @awaelchli @rohitgr7 @carmocca @kaushikb11 @ninginthecloud

### CODE REVIEW

File: pytorch_lightning/loops/optimization/manual_loop.py [hunk 1]
```diff
@@ -146,6 +146,8 @@ def on_run_end(self) -> _OUTPUTS_TYPE:
 
     def _on_before_step(self) -> None:
         self.optim_step_progress.increment_ready()
+        self.trainer.profiler.start("optimizer_step")
```
Comment (carmocca) [inline pytorch_lightning/loops/optimization/manual_loop.py:149]: In automatic optimization, we profile the `optimizer_step` already by calling `_call_lightning_module_hook("optimizer_step")`

And in manual optimization, as shown here

Comment (awaelchli) [reply pytorch_lightning/loops/optimization/manual_loop.py:149]: > This means the profiler entry was duplicated too.

does that warrant a bugfix changelog entry? :)

**Review by awaelchli — APPROVED** [review]

**Review by rohitgr7 — APPROVED** [review]

File: pytorch_lightning/loops/optimization/manual_loop.py (hunk 1 above)
Comment (carmocca) [reply pytorch_lightning/loops/optimization/manual_loop.py:149]: I wouldn't say so. There are no guarantees over the profiler summary format so I wouldn't call it a bugfix. It's a change

**Review by tchaton — APPROVED** [review]
LGTM !

**Pull request merged**